### ISSUE TITLE
Define the DB Schema for `create`d LN Swaps

### ISSUE THREAD

**Issue by D4nte** [comit-network/comit-rs#2173]
## Problem

It is not possible to replicate the existing DB schema model used for rfc003. The main reason is that the messaging, and hence what is stores, differs for HALight. There are no more `swap request`, `accept message` or `decline message`. Instead, the 2 states to save in the DB for a swap are after `create` and `finalize` steps. See https://github.com/comit-network/comit-rs/issues/2134 for details.

This blocked by the REST API because it we want to keep some uniformity across the data types to make our life easier.
To design a data type, we have restrictions from each API.
I believe that the REST API would bring in the most restrictions, hence it would be smoother to drive the design from the REST API to then cascade it to the DB.

Coming with various design in parallel is likely to create unnecessary discrepancies.

## Goal

Design new DB schemas for `create`d HALight swaps. The DB schema must take in account:
- The [new data types](https://github.com/comit-network/comit-rs/issues/2158) that will need to be stored in the DB
- The fact that we want to be able to retrieve swaps which are `create`d but not yet `finalize`d
- Get consensus a name for a "created" swap

The DB Schemas should be defined in diesel and tables created when starting cnd, however, the saving will be done at a later stage (step 3 of https://github.com/comit-network/comit-rs/issues/2134#issuecomment-594308787)

## Out of scope

The DB schema for `finalize`d swap will be done in https://github.com/comit-network/comit-rs/issues/2177.

## Recommendation

KISS.

We want to support han(ether)-halight and halight-herc20 swaps at first. Make sure the database schema can support that. Consider a table design that is split into protocols similar to the body of the API request. This would/should allow composing the tables together for different combinations of swaps.

**Comment by thomaseizinger** [comment]
Please also take into account that approach taken to define those schemas should be expandable to our current swaps as we will migrate the old stuff at some point.

**Comment by thomaseizinger** [comment]
Removing the dependency on https://github.com/comit-network/comit-rs/issues/2158 because the first endpoint was implemented with the walking skeleton and can therefore act as an example.

**Comment by tcharding** [comment]
At some stage the `start_of_swap` will most likely need to be created by the data base i.e., `at DATETIME DEFAULT CURRENT_TIMESTAMP`.  Adding comment here to keep track of it.

`start_of_swap` is mentioned here: https://github.com/comit-network/comit-rs/issues/2408

And this should probably be either done in this ticket or in the [finalized](https://github.com/comit-network/comit-rs/issues/2177) side.

**Comment by D4nte** [comment]
@thomaseizinger you changed the recommendation but this did not go through grooming. I find the recommendation unclear.

**Comment by thomaseizinger** [comment]
Yes sorry!

I changed that together with Philipp yesterday. Should have moved it back into `To-Groom` :grimacing: 

The recommendation is basically what @tcharding did in his draft PR.

**Comment by thomaseizinger** [comment]
Obsoleted by https://github.com/comit-network/comit-rs/issues/2485.
